packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1020,19 +1023,26 @@ export function purposeServiceBuilder(
         purposeSeed.freeOfChargeReason
       );
 
-      const consumerDelegation =
-        await readModelService.getActiveConsumerDelegationByEserviceAndConsumerIds(
-          {
-            eserviceId,
-            consumerId,
-          }
-        );
+      // eslint-disable-next-line functional/no-let
+      let consumerDelegation: Delegation | undefined;
+
+      try {
+        assertRequesterIsConsumer({ consumerId }, authData);
+      } catch {
+        consumerDelegation =
+          await readModelService.getActiveConsumerDelegationByEserviceAndConsumerIds(
+            {
+              eserviceId,
+              consumerId,
+            }
+          );
 
-      assertRequesterCanActAsConsumer(
-        { eserviceId, consumerId },
-        authData,
-        consumerDelegation
-      );
+        assertRequesterIsDelegateConsumer(
```

Comment:
Instead of exporting and using these two validators, we could create a dedicated function for purpose creation that performs the assertions and returns the optional delegationId. It's better to not have these exported so that we don't end up using one of these two instead of assertRequesterCanActAsConsumer

In this way we also avoid the let because we can call a function and assign the return val